File: CS/WebSite/App_Code/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data.Entity.Spatial;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;


    public partial class Category
    {
        public Category()
        {
            Products = new HashSet<Product>();
        }

        public int CategoryID { get; set; }
        [Required]
        [StringLength(15)]
        public string CategoryName { get; set; }
        [Column(TypeName = "ntext")]
        public string Description { get; set; }
        [Column(TypeName = "image")]
        public byte[] Picture { get; set; }
        public virtual ICollection<Product> Products { get; set; }
    }

File: VB/GridEntityFramework/App_Code/Category.vb
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.ComponentModel.DataAnnotations

Namespace GridEntityFramework

    Public Partial Class Category

        Public Sub New()
            Products = New HashSet(Of Product)()
        End Sub

        Public Property CategoryID As Integer

        <Required>
        <StringLength(15)>
        Public Property CategoryName As String

        <Column(TypeName:="ntext")>
        Public Property Description As String

        <Column(TypeName:="image")>
        Public Property Picture As Byte()

        Public Overridable Property Products As ICollection(Of Product)
    End Class
End Namespace
